ui/app/src/utils/content.ts

```diff
@@ -950,3 +954,56 @@ export function getComputedPublishingTarget(item: DetailedItem): PublishingTarge
       ? 'staging'
       : null;
 }
+
+export function cleanupFolderName(name: string, allowBraces: boolean = false): string {
+  let cleanedUpName = slugify(name, {
+    // Setting `strict: true` would disallow `_`, which we don't want.
+    strict: false,
+    // Because of the moment where the library trims, `trim: true` caused undesired replacement of `-`
+    // at the beginning or end of the slug.
+    trim: false
+  });
+  return cleanedUpName.replace(allowBraces ? /[^a-zA-Z0-9-_{}]/g : /[^a-zA-Z0-9-_]/g, '');
+}
+
+export function cleanupAssetName(name: string, allowBraces: boolean = false): string {
+  return name.replace(allowBraces ? /[^a-zA-Z0-9-_{}.]/g : /[^a-zA-Z0-9-_.]/g, '').replace(/\.{1,}/g, '.');
+}
+
+export const getExtension = (type: string, name?: string) => {
+  if (type === 'asset') {
+    const regexFileExtension = /[^\\]*\.(\w+)$/;
+    return name?.match(regexFileExtension)[1];
+  } else {
+    return type === 'controller' ? `groovy` : `ftl`;
+  }
+};
```

Comment:
This is a bit of a dirty mix up. Don't see it being used in multiple places so perhaps it can stay where it was.
The name if should be something like `pickExtensionForItemType`, and the first argument should be typed `SystemType`. Notice there's a util in `path.ts` that gives you the file name extension.

Reply:
this one is used by getName (renamed to getFileNameWithExtensionForItemType), which is used by CreateFileDialogContainer and RenameAssetDialogContainer